packages/dds/test-dds-utils/src/ddsFuzzHarness.ts

```diff
@@ -662,17 +682,43 @@ export function mixinAttach<
 			// However, now that we're transitioning to an attached state, the summarizer client should never have any edits.
 			// Thus we use one of the clients we just loaded as the summarizer client, and keep the client around that we generated the
 			// attach summary from.
-			const summarizerClient = clients[0];
-			clients[0] = state.clients[0];
+			let summarizerClient = clients[0];
+			if (clientA.channel.id === "summarizer") {
+				summarizerClient = clientA;
+			} else {
+				clients[0] = state.clients[0];
+			}
 
 			return {
 				...state,
 				isDetached: false,
 				clients,
 				summarizerClient,
 			};
-		}
+		} else if (operation.type === "rehydrate") {
+			const clientA = state.clients[0];
+			assert.equal(state.clients.length, 1);
+			state.isDetached = true;
 
+			const containerRuntimeFactory = new MockContainerRuntimeFactoryForReconnection(
```

Comment:
I think the problem you're having with id compressor is that the detached container's ids won't be finalized before it's serialized/rehydrated. That happens in the production codepath in ContainerRuntime's `createSummary`.

You can do it here before making a new container runtime factory with:

```
			// This is necessary to get all IdCreationRanges finalized before connecting further clients.
			// The production codepath also finalizes ids before attaching. It's difficult for the mocks
			// to be more direct about this as they don't directly manage any state related to attach
			// (it's up to the user of the mocks to set them up how they want)
			state.containerRuntimeFactory.processAllMessages();
```

(that also duplicates the comment that was added for this step in the attach flow)